projects/soft_999_shicheng_lu_fw_103/src/main.c

```diff
@@ -0,0 +1,54 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "interrupt_def.h"
+#include "log.h"
+#include "wait.h"
+
+// LOG_DEBUG the adc converted value of context pin when called (context is A6 pin)
+void log_adc_a6(const GpioAddress *address, void *context) {
+  GpioAddress *pin_a6_addr = context;
+  uint16_t pin_a6_data = 0;
+  adc_read_converted_pin(*pin_a6_addr, &pin_a6_data);
+  LOG_DEBUG("adc converted pin A6 data: %d\n", pin_a6_data);
+}
+
+int main(void) {
+  interrupt_init();
+  gpio_init();
+  adc_init(ADC_MODE_SINGLE);
+
+  // output gpio pin A6
+  GpioAddress analog_pin_a6_addr = { .port = GPIO_PORT_A, .pin = 6 };

```

Comment:
This is a bit nitpicky, but you should place a comma after the last value (ie place a comma after the '6'.
This will allow the formater to fix the formatting of this and change it to look something like:
```
GpioAddress analog_pin_a6_addr = {
    .port = GPIO_PORT_A,
    .pin = 6,
  };
```

Also the naming is sort of inconsistent since you name this pin `analog_pin_a6_addr`, but then later for your button pin, you simply name it `button_addr`. Please try to keep it consistent, I would probably just rename 
`analog_pin_a6_addr` to something like 'adc_addr'